### ISSUE TITLE
stack overflow on importing (possibly large) corpus

### ISSUE THREAD

**Issue by Herrner** [korpling/graphANNIS#205]
**Describe the bug**
`annis import` aborts on importing a large corpus.
```
08:12:34 [INFO] loaded 7400000 lines from relANNIS_1.1/relAnnis_anno/node.annis
08:12:41 [INFO] loaded 7500000 lines from relANNIS_1.1/relAnnis_anno/node.annis
08:12:49 [INFO] loaded 7600000 lines from relANNIS_1.1/relAnnis_anno/node.annis

thread 'main' has overflowed its stack
fatal runtime error: stack overflow
Aborted (core dumped)
```
**To Reproduce**
1. start ANNIS in the data directory `~/annis /mnt/annis/v4/`
2. import relAnnis-Corpus (either with `>> import relANNIS_1.1/relAnnis_anno/` which is the unpacked directory or with `>> import relANNIS_1.1_packed.zip` which is the compressed version, which can be found [here](https://www.fdr.uni-hamburg.de/record/9195/files/relANNIS_1.1.zip?download=1))
3. ANNIS starts importing and exits after loading 7600000 lines

**Expected behavior**
The corpus import should run through.

**Desktop (please complete the following information):**
- OS: Ubuntu Linux 20.04 LTS
- Java Version: 
  ```
  openjdk version "17" 2021-09-14
  OpenJDK Runtime Environment (build 17+35-Ubuntu-120.04)
  OpenJDK 64-Bit Server VM (build 17+35-Ubuntu-120.04, mixed mode, sharing)
   ```
 - ANNIS-Version: `graphANNIS CLI 1.3.0`

To rule out file storage limitations, I moved the data folder to a separate partition (hence the `/mnt/annis/v4/`) - that did not do the trick.

**Comment by thomaskrause** [comment]
I was able to find several issues in the relANNIS import that caused unexpected main memory usage. Some of them were previously made trade-offs between "speed vs. main memory" which I tuned more in direction of "less main memory". However, even when using the disk-based import, the import of larger corpora will need about 4 GB of main memory (but less when the corpus is imported because we don't need some of the expensive temporary calculations).

The memory consumption was especially problematic for this corpus because it uses a lot of escape characters in the "node.annis" table and this triggered a weird memory fragmentation issue.

All these issues are addressed in #206 and as soon as I can finish more extensive tests for possible speed regressions I will make a new release of graphANNIS and ANNIS with these fixes.

**Comment by thomaskrause** [comment]
Fixed most memory issues, so the corpus only needs about 4GB during import (much less when just querying it). Still, the process is much too slow in my taste, with almost 2 hours import time on my notebook. I will try to adress these in seperate pull requests.

**Comment by thomaskrause** [comment]
@Herrner Fix was released in graphANNIS 1.4.0 and ANNIS 4.5.3
For the specific corpus my notebook still needs about 1.5 hours to import, but with the disk-based mode the needed main memory is at most 4 GB during import. If you have more main memory available on the server, the default main memory mode for the corpus needs around 6GB in production but enables much faster queries. Especially for "dumb" but costly queries like `tok`, that produce large output sizes are about ten times faster in main memory mode (as expected). For more specific queries with a limited result size the difference is not that large.
